### PR TITLE
Add specs for attribute selectors with modifiers

### PR DESCRIPTION
**Pull request by nex3**
See sass/dart-sass#656

[skip dart-sass]

### CODE REVIEW

File: spec/css/selector.hrx [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+
+   ---^
+
+<===>
+================================================================================
+<===> error/attribute/modifier/unicode/input.scss
+// Attribute modifiers must be ASCII alphabetical characters.
+[a=b ï] {c: d}
+
+<===> error/attribute/modifier/unicode/error
```
Comment (jathak) [inline spec/css/selector.hrx]: It looks like this is failing becomes of a missing error message here.

Comment (nex3) [reply spec/css/selector.hrx]: Fixed, thanks!

File: lib/sass_spec/util.rb [hunk 2]
```diff
@@ -3,7 +3,7 @@ class << self
     # Normalizes the whitespace in the given CSS to make it easier to compare
     # across implementations.
     def normalize_output(css)
-      css.gsub(/(?:\r?\n)+/, "\n")
+      css.gsub(/(?:\r?\n)+/, "\n").force_encoding("utf-8")
```
Comment (glebm) [inline lib/sass_spec/util.rb]: `force_encoding` doesn't re-encode, it simply forces Ruby to interpret them as UTF-8.
If they're already UTF-8, it shouldn't be necessary. If they're not UTF-8, this won't help (you'd want `encode('utf-8')` instead).

What was the issue here?

Comment (nex3) [reply lib/sass_spec/util.rb]: Both [Dart Sass](https://github.com/sass/sass-spec/blob/6a11df8b633114b595eab96d831b31646aaa5661/lib/sass_spec/engine_adapter.rb#L97-L98) and [LibSass](https://github.com/sass/sass-spec/blob/6a11df8b633114b595eab96d831b31646aaa5661/lib/sass_spec/engine_adapter.rb#L46) explicitly set their output streams to binary encoding. I believe this is necessary for some of the LibSass regression tests involving invalid UTF-8-encoded input.

**Pull request merged**